src/test/scala/io/iohk/ethereum/blockchain/sync/StateSyncSpec.scala

```diff
@@ -89,6 +91,24 @@ class StateSyncSpec
     }
   }
 
+  it should "start state sync when receiving start signal while bloom filter is loading" in new TestSetup() {
+    override def buildBlockChain(): BlockchainImpl = {
+      val storages = getNewStorages
+      //iterating 1M key and values should force scheduler actor o enqueue last received command i.e StartSyncing
+      (0 until 1000000).foreach { i =>
```

Comment:
what about overriding storages in a way that will make source for bloom filter return observable which doesn't emit elements until some signal?